.travis.yml

```diff
@@ -137,64 +163,80 @@ install:
   # Building with tests and benchmarks...
   # build & run tests, build benchmarks
-  - ${CABAL} v2-build -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - ${CABAL} v2-build $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output
   # Testing...
-  - ${CABAL} v2-test -w ${HC} ${TEST} ${BENCH} all | color_cabal_output
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then ${CABAL} v2-test $WITHCOMPILER ${TEST} ${BENCH} all | color_cabal_output ; fi
   # Doctest...
-  - "doctest --fast $(find */src  -name '*.hs' ! -path 'template-haskell-optics*/*') -XBangPatterns -XDefaultSignatures -XFlexibleContexts -XFlexibleInstances -XFunctionalDependencies -XDeriveFunctor -XGADTs -XLambdaCase -XMultiParamTypeClasses -XRankNTypes -XScopedTypeVariables -XTupleSections -XTypeFamilies -XTypeOperators -XDataKinds -XTypeApplications -XInstanceSigs"
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics} && doctest  src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_core} && doctest  -XBangPatterns -XDefaultSignatures -XDeriveFunctor -XFlexibleContexts -XFlexibleInstances -XFunctionalDependencies -XGADTs -XInstanceSigs -XLambdaCase -XMultiParamTypeClasses -XRankNTypes -XScopedTypeVariables -XTupleSections -XTypeApplications -XTypeFamilies -XTypeOperators src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_extra} && doctest  -XBangPatterns -XDefaultSignatures -XDeriveFunctor -XFlexibleContexts -XFlexibleInstances -XFunctionalDependencies -XGADTs -XInstanceSigs -XLambdaCase -XMultiParamTypeClasses -XRankNTypes -XScopedTypeVariables -XTupleSections -XTypeApplications -XTypeFamilies -XTypeOperators src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_sop} && doctest  src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_th} && doctest  src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_vl} && doctest  src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_generic_optics} && doctest  src) ; fi
+  - if ! $GHCJS && [ $HCNUMVER -lt 80800 ] ; then (cd ${PKGDIR_template_haskell_optics} && doctest  src) ; fi
   # cabal check...
-  - (cd optics-* && ${CABAL} -vnormal check)
-  - (cd optics-core-* && ${CABAL} -vnormal check)
-  - (cd optics-extra-* && ${CABAL} -vnormal check)
-  - (cd optics-sop-* && ${CABAL} -vnormal check)
-  - (cd optics-th-* && ${CABAL} -vnormal check)
-  - (cd optics-vl-* && ${CABAL} -vnormal check)
-  - (cd generic-optics-* && ${CABAL} -vnormal check)
-  - (cd template-haskell-optics-* && ${CABAL} -vnormal check)
-  - (cd metametapost-* && ${CABAL} -vnormal check)
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics} && ${CABAL} -vnormal check) ; fi
+  - (cd ${PKGDIR_optics_core} && ${CABAL} -vnormal check)
+  - if ! $GHCJS && [ $HCNUMVER -lt 80802 ] ; then (cd ${PKGDIR_optics_extra} && ${CABAL} -vnormal check) ; fi
```

Comment:
hmm. let me fix this uglyness.

we don't need upper bound, as we don't test with GHC HEAD, should we?

Reply:
Compilation with 8.8.1 used to work: https://travis-ci.org/well-typed/optics/builds/565907397